docs/getting_started/go/hello_world_in_go/index.md

```diff
@@ -83,419 +85,767 @@ go get go.temporal.io/sdk
 
-:::
 
-The function defines the options to execute an Activity, and then executes an Activity called `ComposeGreeting`, which you'll define next. The function returns the result of the Activity.
+This code does check for and handle errors, but because you've configured unlimited retries, there won't be any exceptions caught. However, if you change the Retry Policy's maximum retries, or you specify non-retryable exceptions, this code will be in place to handle those errors.
+
+Next you'll create a Worker that executes the Workflow and Activities.
+
+## Configure and run a Worker
+
+When you start a Temporal Workflow, the Workflow and its Activities get scheduled on the Temporal Service's [Task Queue](https://docs.temporal.io/concepts/what-is-a-task-queue). A [Worker](https://docs.temporal.io/concepts/what-is-a-worker) hosts Workflow and Activity functions and polls the Task Queue for tasks related to those Workflows and Activities. After the Worker runs the code, it communicates the results back to the Temporal Service where they're stored in the Event History. This records the Workflow's entire execution, enabling features like fault tolerance by allowing the Workflow to replay in case of Worker crashes or restarts.
 
-With your Workflow Definition created, you're ready to create the `ComposeGreeting` Activity.
+You use the Temporal SDK to define a Worker Program.
 
-## Create an Activity
+In your Worker Program, you need to specify the name of the Task Queue, which must match the Task Queue name used whenever you interact with a Workflow from a client application. This ensures that the Worker processes tasks for the correct Workflow. The Task Queue name is a case-insensitive string. To ensure consistency and avoid errors, define the Task Queue name as a constant that can be reused throughout your code.
 
-In a Temporal Application, Activities are where you execute [non-deterministic](https://docs.temporal.io/workflows#deterministic-constraints) code or perform operations that may fail, such as API requests or database calls. Your Workflow Definitions call Activities and process the results. Complex Temporal Applications have Workflows that invoke many Activities, using the results of one Activity to execute another.
+Create the file `shared.go`:
 
-For this tutorial, your Activity won't be complex; you'll create an Activity that takes a string as input and uses it to create a new string as output, which is then returned to the Workflow. This will let you see how Workflows and Activities work together without building something complicated.
+```
+touch shared.go
+```
 
-With the Temporal Go SDK, you define Activities similarly to how you define Workflows: using a regular exportable Go function.
+Open the file and add the following lines to the file to define the constant for the Task Queue:
 
-Create the file `activity.go` in the project root and add the following code to define a `ComposeGreeting` function:
+<!--SNIPSTART go-ipgeo-shared-->
```

Comment:
Why Typescript here? I don't think it actually makes a difference, but it looks like an accidental holdover from the TS version.

Reply:
Fixed. This was a weird snipsync edge case.